x/rollapp/keeper/keeper.go

```diff
@@ -29,28 +32,71 @@ func NewKeeper(
 	storeKey,
 	memKey storetypes.StoreKey,
 	ps paramtypes.Subspace,
-	ibcclientkeeper types.IBCClientKeeper,
+	ibcclientKeeper types.IBCClientKeeper,
+	channelKeeper types.ChannelKeeper,
+	bankKeeper types.BankKeeper,
 ) *Keeper {
 	// set KeyTable if it has not already been set
 	if !ps.HasKeyTable() {
 		ps = ps.WithKeyTable(types.ParamKeyTable())
 	}
 
 	return &Keeper{
-
 		cdc:             cdc,
 		storeKey:        storeKey,
 		memKey:          memKey,
 		paramstore:      ps,
 		hooks:           nil,
-		ibcclientkeeper: ibcclientkeeper,
+		ibcclientKeeper: ibcclientKeeper,
+		channelKeeper:   channelKeeper,
+		bankKeeper:      bankKeeper,
 	}
 }
 
 func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
+// TriggerGenesisEvent triggers the genesis event for the rollapp.
+func (k Keeper) TriggerRollappGenesisEvent(ctx sdk.Context, rollapp types.Rollapp) error {
```

Comment:
I would keep this method private and move it, along with 'mintRollappGenesisTokens' to a new file: msg_server_trigger_genesis_event.go